### PR TITLE
Add boundary perftest

### PR DESCRIPTION
**Pull request by lukasm91**
- Fix/complete existing boundary conditions test
- use perftest structure for boundary conditions test
- Factor out common code of boundary conditions and layout transformation perftest

After #1311

### CODE REVIEW

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch icgrid

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch icgrid

**Comment by lukasm91** [comment]
launch icgrid

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
do perftest

File: regression/boundary_conditions.cpp [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        GT_FUNCTION void operator()(
+            Direction, DataField0 &data_field0, DataField1 const &data_field1, uint_t i, uint_t j, uint_t k) const {
+            data_field1(i, j, k) = data_field0(i, j, k) * value;
+        }
+
+        // relative coordinates
+        template <sign I, sign K, typename DataField0, typename DataField1>
+        GT_FUNCTION void operator()(
+            direction<I, minus_, K>, DataField0 &, DataField1 const &data_field1, uint_t i, uint_t j, uint_t k) const {
+            data_field1(i, j, k) = 88 * value;
```
Comment (havogt) [inline regression/boundary_conditions.cpp]: Up to you: I don't like this hard-coded random numbers. Consider giving them a name. (I know it is copied from the old example...)

Comment (lukasm91) [reply regression/boundary_conditions.cpp]: done

**Comment by lukasm91** [comment]
launch strgrid
